### PR TITLE
Restaurant Index Api Response

### PR DESCRIPTION
**Pull request by thesuss**
PT Story: 
https://www.pivotaltracker.com/story/show/131984201

Changes proposed in this pull request:
- Add request specs for Restaurant api
- Add controller for Restaurant api
- Add jbuilder file for Restaurant index api response

What I have learned working on this feature: 
- An error in Routes can change the expected response from the application
- API:s are rather easy to create, with the tools provided

Screenshots (postman):
skarmavbild 2016-10-10 kl 20 03 34

Ready for review!

### CODE REVIEW

File: spec/requests/api/v1/restaurants_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::RestaurantsController, type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  describe 'GET /v1/restaurants' do
+    let!(:owner) do
+      create(:user, email: 'whatever@random_restaurant.com', role: 'owner')
+    end
+    let!(:owner_2) do
```
Comment (tochman) [inline spec/requests/api/v1/restaurants_spec.rb]: This one is never used.

File: spec/requests/api/v1/restaurants_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      create(:user, email: 'whatever@random_restaurant2.com', role: 'owner')
+    end
+    let!(:restaurant) { create(:restaurant, user: owner) }
+    let!(:restaurant_2) { create(:restaurant, user: owner) }
+
+    it 'should return json with restaurants' do
+      get '/api/v1/restaurants'
+      json_response = JSON.parse(response.body)
```
Comment (tochman) [inline spec/requests/api/v1/restaurants_spec.rb]: Could be extracted to a separate method or helper since you're using this in many tests

**Review by tochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/restaurants_spec.rb (hunk 1 above)
Comment (thesuss) [reply spec/requests/api/v1/restaurants_spec.rb]: Removed

File: spec/requests/api/v1/restaurants_spec.rb (hunk 2 above)
Comment (thesuss) [reply spec/requests/api/v1/restaurants_spec.rb]: Done

File: spec/requests/api/v1/restaurants_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::RestaurantsController, type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
```
Comment (tochman) [inline spec/requests/api/v1/restaurants_spec.rb]: Where are you using the headers?

File: spec/support/response_json.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module ResponseJSON
+  def response_json
+    JSON.parse(response.body)
+  end
+end
+
+RSpec.configure do |config|
+  config.include ResponseJSON
+end
```
Comment (tochman) [inline spec/support/response_json.rb]: I would add this config in rails_helper for better readability

File: spec/requests/api/v1/restaurants_spec.rb (hunk 3 above)
Comment (thesuss) [reply spec/requests/api/v1/restaurants_spec.rb]: removed

File: spec/support/response_json.rb (hunk 4 above)
Comment (thesuss) [reply spec/support/response_json.rb]: done

**Review by tochman — APPROVED** [review]

**Comment by tochman** [comment]
So there are issues with this RP that will lead to merge conflicts in the future, but let's go with that.

**Pull request merged**